homeassistant/components/homeconnect/binary_sensor.py

```diff
@@ -0,0 +1,72 @@
+"""Provides a binary sensor for Home Connect.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/integrations/binary_sensor.homeconnect/
+"""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+from .api import HomeConnectEntity
+from .const import DEVICES, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Home Connect binary sensor."""
+
+    def get_entities():
+        entities = []
+        data = hass.data[DOMAIN]
+        for device_dict in data.get(DEVICES, []):
```

Comment:
This seems like a bug as you write to `hass.data[DOMAIN]conig_entry.entry_id]` in the component.